environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/EnvCreationActions.java

```diff
@@ -87,11 +87,12 @@ protected void doExecute(CommonContext context, EnvCreationEvent payload, Map<Ob
             protected void doExecute(CommonContext context, ResourceCrnPayload payload, Map<Object, Object> variables) {
                 environmentService
                         .findEnvironmentById(payload.getResourceId())
-                        .ifPresent(environment -> {
+                        .ifPresentOrElse(environment -> {
                             environment.setStatusReason(null);
                             environment.setStatus(EnvironmentStatus.AVAILABLE);
                             environmentService.save(environment);
-                        });
+                        }, () -> LOGGER.error("Cannot finish the creation of env, because the environment does not exists: {}. "
+                                + "But the flow will continue, how can this happen?", payload.getResourceId()));
                 LOGGER.info("ENV_CREATION_FINISHED_STATE");
```

Comment:
These lines in each action could also be changed to be something more meaningful